adafruit_tinylora/adafruit_tinylora.py

```diff
@@ -279,17 +281,18 @@ def send_packet(self, lora_packet, packet_length):
             k += 1
         # switch RFM to TX operating mode
         self._write_u8(_REG_OPERATING_MODE, _MODE_TX)
-        # wait for TxDone IRQ
         print('Sending packet')
-        send_attempt = 0
-        while not self._irq.value and send_attempt < 15:
-            # waiting for TxDone
-            time.sleep(1)
-            send_attempt += 1
+        # wait for TxDone IRQ, poll for timeout.
+        start = time.monotonic()
+        timed_out = False
+        while not timed_out and not self._irq.value:
+          if(time.monotonic() - start) >= timeout:
+            timed_out = True
         # switch RFM to sleep operating mode
         print('Packet Sent!')
         self._write_u8(_REG_OPERATING_MODE, _MODE_SLEEP)
 
```

Comment:
If the transmit timed out should an Error be thrown??  If so, it should occur after going back to sleep.
something like https://github.com/adafruit/Adafruit_CircuitPython_RFM9x/blob/master/adafruit_rfm9x.py#L579

Reply:
Yep, wasn't sure where to throw the error. After it sleeps is the best, since it "resets" the RFM's state.

Tested with a small timeout and code threw: `RuntimeError: Timeout during packet send`